tests/cpp/test_cdpruner_dpp.cpp

```diff
@@ -0,0 +1,523 @@
+    auto text_features = createTextFeatures(batch_size, hidden_dim, 0.5f, 0.1f);
+
+    // Test multi-frame pruning using vector interface
+    auto pruned_features = cdpruner.apply_pruning(visual_features, text_features);
+    auto pruned_shape = pruned_features.get_shape();
+
+    // Verify output dimensions
+    EXPECT_EQ(pruned_shape.size(), 3) << "Pruned features should have 3 dimensions [batch, sequence, hidden]";
+    EXPECT_EQ(pruned_shape[0], batch_size) << "Batch size should remain unchanged";
+    EXPECT_EQ(pruned_shape[2], hidden_dim) << "Hidden dimension should remain unchanged";
+
+    // Verify sequence length is reduced (pruning applied)
+    size_t total_original_length = num_frames * sequence_length_per_frame;
+    EXPECT_LT(pruned_shape[1], total_original_length) << "Sequence length should be reduced after pruning";
+    EXPECT_GT(pruned_shape[1], 0) << "Pruned sequence should have at least some tokens";
+
+    // Verify pruning ratio is approximately correct (allowing some tolerance)
+    double actual_pruning_ratio = 1.0 - (double)pruned_shape[1] / total_original_length;
+    double expected_ratio = cdp_config.pruning_ratio / 100.0;
+    double tolerance = 0.1;  // 10% tolerance for pruning ratio
+    EXPECT_NEAR(actual_pruning_ratio, expected_ratio, tolerance)
+        << "Actual pruning ratio should be close to configured ratio";
+
+    // Verify that output tensor is valid and contains reasonable values
+    EXPECT_FALSE(pruned_features.get_shape().empty()) << "Pruned features should not be empty";
+    EXPECT_GT(pruned_features.get_byte_size(), 0) << "Pruned features should have non-zero size";
+}
+
+TEST_P(CDPrunerIntegrationTest, MultiFramePruningEdgeCases) {
+    // Test edge cases for multi-frame pruning
+    CDPruner cdpruner(cdp_config);
+
+    size_t hidden_dim = 128;
+    size_t batch_size = 1;
+
+    // Test case 1: Single frame (should work like regular pruning)
+    {
+        std::vector<ov::Tensor> single_frame;
+        ov::Tensor frame_features(ov::element::f32, {batch_size, 50, hidden_dim});
+        initializeTensorWithPattern(frame_features, 0.5f, 0.01f);
+        single_frame.push_back(std::move(frame_features));
+
+        auto text_features = createTextFeatures(batch_size, hidden_dim);
+
+        auto result = cdpruner.apply_pruning(single_frame, text_features);
+        EXPECT_FALSE(result.get_shape().empty()) << "Single frame pruning should produce valid result";
+        EXPECT_LT(result.get_shape()[1], 50) << "Single frame should be pruned";
+    }
+
+    // Test case 2: Empty frames vector (should handle gracefully)
+    {
+        std::vector<ov::Tensor> empty_frames;
+        auto text_features = createTextFeatures(batch_size, hidden_dim);
+
+        // This should either return empty tensor or handle gracefully
+        auto result = cdpruner.apply_pruning(empty_frames, text_features);
+        // We don't expect this to crash - specific behavior depends on implementation
```

Comment:
Why not `EXPECT_NO_THROW(result = cdpruner.apply_pruning(empty_frames, text_features));` ?